### ISSUE TITLE
Host fee amounts not matching up

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#6397]
I'm trying to do year-end financial reports for last year (1 Apr 2021 - 31 Mar 2022) for OCNZ, and I'm having trouble nailing down the total received in Host fees.

If I go to host admin reports and set this date range, it says $3488.21
<img width="310" alt="Screen Shot 2023-02-08 at 1 41 21 PM" src="https://user-images.githubusercontent.com/5498878/217401542-ee5589eb-d4ba-4bbc-8aed-212562183394.png">

If I download a CSV of all transactions and add up all the one with type HOST_FEE I get $1201.80.

If I go to the Host's transactions, set the date range and filter by Host Fee credits ([this report](https://opencollective.com/ocnz/transactions?kind=HOST_FEE&period=2021-03-31T11%3A00%3A00.000Z%E2%86%922022-03-31T10%3A59%3A59.999Z)) and download that CSV and total it, I get $1987.03.

If I look at total incoming contributions for that period (by downloading the CSV version of the chart on Host admin reports), it's $97241.84 which multiplied by the 5.75% host fee is $5591.41.

Basically, I have to report this number for tax purposes and I'm a bit lost on how to nail it down accurately.

**Comment by znarf** [comment]
The new ledger with separated HOST_FEE transactions is only available from June 1st 2021 UTC

From June 1st 2021 to April 1st 2022 I confirm you should see: $1987.03 NZD

Reference:

```
SELECT SUM(t1."amountInHostCurrency") as "_amount", t1."hostCurrency" as "_currency"
FROM "Transactions" as t1
WHERE t1."HostCollectiveId" = 125178
AND t1."CollectiveId" = 125178
AND t1."kind" = 'HOST_FEE'
AND t1."createdAt" >= '2021-04-01 00:00:00+00'
AND t1."createdAt" <= '2022-04-01 00:00:00+00'
AND t1."deletedAt" IS NULL
GROUP BY t1."hostCurrency"
```

> If I download a CSV of all transactions and add up all the one with type HOST_FEE I get $1201.80.

Maybe the export is truncated and you are missing transactions?

> If I go to host admin reports and set this date range, it says $3488.21

There is $1501.18 NZD in Host Fee collected between April 1st 2021 and June 1st 2021. Reported in the ledger using the `hostFeeInHostCurrency` column. I can confirm this number too.

Reference:

```
SELECT ABS(SUM(t1."hostFeeInHostCurrency")) as "_amount", t1."hostCurrency" as "_currency"
FROM "Transactions" as t1
WHERE t1."HostCollectiveId" = 125178
AND t1."createdAt" >= '2021-04-01 00:00:00+00'
AND t1."createdAt" <= '2022-06-01 00:00:00+00'
AND NOT (t1."type" = 'DEBIT' AND t1."kind" = 'ADDED_FUNDS')
AND t1."deletedAt" IS NULL
GROUP BY t1."hostCurrency"
```

**Comment by alanna** [comment]
@znarf so $3488.21 NZD is the correct number overall?

**Comment by znarf** [comment]
Yes, I can confirm the $3488.21 NZD number.

**Comment by alanna** [comment]
> Maybe the export is truncated and you are missing transactions?

Oh no, how do I get an un-truncated CSV @znarf ? I need the full data to file the Host's taxes.

**Comment by znarf** [comment]
For export up to 2 months there is normally no limit. Out of this, there is a default of 1000 transactions, we have to do this to protect from abuse. To skip the limit:
https://rest.opencollective.com/v2/ocnz/hostTransactions.csv?dateFrom=2021-04-01T00%3A00%3A00.000Z&dateTo=2022-03-31T23%3A59%3A59.999Z&fetchAll=1

**Comment by iamronen** [comment]
@alanna I would like to echo back my understanding of what I understand you are trying to do and where things went wrong  and  I would appreciate your feedback on my understanding.

1. As part of your "end of year" process for OCNZ you are looking for a number "total host fees.
2. You found this number in the report available to you as fiscal host admin.
3. But you don't completely trust it and so you want to try somehow confirm this number by cross-referencing it with another source. If you get the same result you feel more confident with the number you have. If you get a different result your confidence is diminished.
4. So You tried testing with two other sources.
5. The first source you turned to was the transactions download. This didn't match for two reasons:
     - There was a change in the ledger in June 2021 in how host fees are recorded.
     - You thought you were downloading a complete dataset when in reality you were not.
 6. The second data source you turned to was the "total incoming contributions and you tried applying to that the host fee. I am unclear why this one did not match but to my understanding real host fees can vary (sometimes waved, sometimes reduced, etc.) and so this method is an incorrect approach to verification. Is this correct?

I am left with these questions: 
1. Why did you not trust the initial result on the screen?
2. What would it take for the platform to earn that trust?
3. Why did you expect the second verification method to work (or is my understanding wrong about its validity)?

**Comment by iamronen** [comment]
@znarf 
1. When we achieve on the platform a separation between workplace & public will it then be possible to remove this limitation inside the workspace (where a known legitimate user is asking for the information)? 
2. Is there a reason not to indicate on the public CSV download that there is a download limit of 1000 transactions with an indication if the current dataset exceeds the limit?

**Comment by alanna** [comment]
@znarf would it be possible to automatically lift this limitation on data exports for trusted users, first of all platform admins, but ideally admins of trusted hosts? I have a feeling this will cause very painful issues with accounting for people if they don't realise it's truncated. We should definitely put a big warning when you download the CSV.

**Comment by alanna** [comment]
@iamronen it's no so much that I didn't trust the figure on the host report screen (which does tend to be accurate and ended up being correct). it's that this report screen does not give me most of the data outputs I need to do things like taxes and year-end financials for OCNZ as a company, so I have to work with the raw data. So I exported it and worked with on a spreadsheet with my accountant, to get figures like total expenses, GST stuff, total income, etc. My accountant looked at the total of host fees she could see in that raw data and noted it didn't match up with the $3488.21 NZD amount, and I had to figure out why. I'm trying to run OCNZ without using a separate accounting software, and only rely on OC data for accounting stuff, and it's pretty messy.
